chapters/VK_EXT_debug_utils.txt

```diff
@@ -66,17 +74,6 @@ calling fname:vkSetDebugUtilsObjectNameEXT again with a new string.
 If pname:pObjectName is an empty string, then any previously set name is
 removed.
 
-.Valid Usage
-****
-  * [[VUID-VkDebugUtilsObjectNameInfoEXT-objectType-01905]]
-    pname:objectType must: not be ename:VK_OBJECT_TYPE_UNKNOWN
-  * [[VUID-VkDebugUtilsObjectNameInfoEXT-objectHandle-01906]]
-    pname:objectHandle must: not be dlink:VK_NULL_HANDLE
-  * [[VUID-VkDebugUtilsObjectNameInfoEXT-objectHandle-01907]]
-    pname:objectHandle must: be a Vulkan object of the type associated with
-    pname:objectType as defined in <<debugging-object-types>>.
```

Comment:
Should still say "if not unknown type, must be valid handle". It might come up in some obscure case, e.g. `api_dump` layer after `unique_objects` layer -- i.e. the handle must be properly translatable without crashes.

Reply:
How about this:

```
  * If pname:objectType is not ename:VK_OBJECT_TYPE_UNKNOWN,
    pname:objectHandle must: be a Vulkan handle of the type associated with
    pname:objectType as defined in the
    <<debugging-object-types, VkObjectType and Vulkan Handle Relationship>>
    table
```
That would allow me to remove the VU statement we just added at `vkSetDebugUtilsObjectNameEXT` since that covers this case.  The only VU statement left for `vkSetDebugUtilsObjectNameEXT` would be this:

```
  * pname:pNameInfo\->pname:objectType must: not be ename:VK_OBJECT_TYPE_UNKNOWN
```

Since we don't want UNKNOWN there.

Reply:
Wait I am starting to get lost. Let's recapitulate.

There are two users of `VkDebugUtilsObjectNameInfoEXT`:  `vkSetDebugUtilsObjectNameEXT` and `VkDebugUtilsMessengerCallbackDataEXT` (which in turn is used only by `vkSubmitDebugUtilsMessageEXT`; and by user's callback but that does not count towards VU).

For `vkSetDebugUtilsObjectNameEXT` we always want the and only valid type **and** valid handle.

For `vkSubmitDebugUtilsMessageEXT` I think we agreed it should be valid to provide:
- valid type + valid handle
- valid type + null handle

(
The invalid usage for `vkSubmitDebugUtilsMessageEXT` therefore should be:
- unknown type
- valid type + invalid non-null handle

)

That about right?

<hr />

So if we do GCD, then `VkDebugUtilsObjectNameInfoEXT` can have the
"must not be `UNKNOWN_TYPE`" VU,

The `vkSetDebugUtilsObjectNameEXT` then needs
"`object` must be valid `objectType` handle" VU.

And the `vkSubmitDebugUtilsMessageEXT` would need
"If `object` is not NULL, `object` must be valid `objectType` handle" VU.

(Which alternatively could be moved to `VkDebugUtilsObjectNameInfoEXT`, and then `vkSetDebugUtilsObjectNameEXT` would not need to note the handle type: "`object` must be a valid Vulkan handle")

Reply:
We might have `VK_OBJECT_TYPE_UNKNOWN` type for `VkDebugUtilsObjectNameInfoEXT` when it's returned in the callback data if the layers fail to determine a handle in some cases.  This can be the case when an extension adds a type and isn't implemented in the layers (but is in the driver). 

So, I don't think "must not be UNKNOWN_TYPE" VU, belongs on `VkDebugUtilsObjectNameInfoEXT`.  Instead, I think it needs to be called out during the two commands.

This is the way I see it:

`VkDebugUtilsObjectNameInfoEXT`
  * If type == unknown, handle must not be NULL

`vkSubmitDebugUtilsMessageEXT`
  * For each object in pCallbackData->pObjects, type must: not be Unknown
  * For each object in pCallbackData->pObjects, if handle not NULL must be a valid handle

`vkSetDebugUtilsObjectNameEXT`
 * pNameInfo->objectType must: not be Unknown
 * pNameInfo->objectHandle must: be a valid handle

Reply:
Hmm, I never considered VUs to apply to stuff the Implementation returns (or calls calbacks with)...
I think returned values (and parameters from Implementation to the Applications callback) needs its own restrictions clerly addressed to the Implementers.

Reply:
`vkSubmitDebugUtilsMessageEXT` is stricter because the application should know everything about the objects that underlying layers/ICDs may not.  That way, we force them to pass in useful data.

Reply:
Then again, why should we care what the app passes to its own callback? Especially if the callback must already be able to handle the unrestricted cases.

Besides "application" may be just another layer. layer 2 (some library) may not know much more than the layers and ICD does.

And it is also weird to me that the layers/ICDs would not know. Either they do not know a thing is a handle\object in the first place, or they do know the type, right?